File: backend/app/schemas/reminders.py
"""Pydantic schemas for reminder endpoints."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class ReminderCreateRequest(BaseModel):
    title: str = Field(..., max_length=200)
    due_at: datetime
    timezone: str | None = None


class ReminderUpdateRequest(BaseModel):
    title: str | None = Field(None, max_length=200)
    due_at: datetime | None = None


class ReminderResponse(BaseModel):
    id: str
    call_id: str | None = None
    title: str
    due_at: datetime
    timezone_at_creation: str | None = None
    status: str
    created_at: datetime
    call_from_masked: str | None = None

    model_config = {"from_attributes": True}


class ReminderListResponse(BaseModel):
    items: list[ReminderResponse]

File: mobile/src/api/reminders.ts
import { apiClient } from './client';

export interface Reminder {
  id: string;
  call_id: string | null;
  title: string;
  due_at: string;
  timezone_at_creation: string | null;
  status: string;
  created_at: string;
  call_from_masked: string | null;
}

export interface ReminderCreateParams {
  title: string;
  due_at: string;
  timezone?: string;
}

export interface ReminderUpdateParams {
  title?: string;
  due_at?: string;
}

export async function listReminders(): Promise<{ items: Reminder[] }> {
  const { data } = await apiClient.get('/reminders');
  return data;
}

export async function createReminder(
  callId: string,
  params: ReminderCreateParams,
): Promise<Reminder> {
  const { data } = await apiClient.post(`/reminders/calls/${callId}`, params);
  return data;
}

export async function updateReminder(
  reminderId: string,
  params: ReminderUpdateParams,
): Promise<Reminder> {
  const { data } = await apiClient.patch(`/reminders/${reminderId}`, params);
  return data;
}

export async function completeReminder(reminderId: string): Promise<Reminder> {
  const { data } = await apiClient.post(`/reminders/${reminderId}/complete`);
  return data;
}

export async function cancelReminder(reminderId: string): Promise<Reminder> {
  const { data } = await apiClient.post(`/reminders/${reminderId}/cancel`);
  return data;
}

export async function deleteReminder(reminderId: string): Promise<void> {
  await apiClient.delete(`/reminders/${reminderId}`);
}
